unity/PingPongGameExample/PingPongGame/Assets/Scripts/GameManager.cs

```diff
@@ -140,7 +140,30 @@ async Task Start()
       roomIdInput = GameObject.Find("InputFieldRoomId").GetComponent<InputField>();
       roomIdInput.onEndEdit.AddListener(ConnectToServerWithRoomId);
 
+      yield return CheckLocationIsRunning();
```

Comment:
yield return StartCoroutine(CheckLocationIsRunning());
will start the Coroutine before the yield and wait for it, I think this what u intending to do

Reply:
What is the difference in behavior between StartCoroutine(CheckLocationIsRunning()); and CheckLocationIsRunning();?

Reply:
The difference is one frame , 

yield return CheckLocationIsRunning(); > yield return null (one frame) >> StartCoroutine(CheckLocationIsRunning()) >>  continue

yield return StartCoroutine(CheckLocationIsRunning()) > StartCoroutine(CheckLocationIsRunning()) >> continue

<img width="562" alt="Screen Shot 2020-07-06 at 10 57 37 AM" src="https://user-images.githubusercontent.com/24863504/86624086-9892ea80-bf77-11ea-8e32-8ce97b0ed98e.png">
[link](https://docs.unity3d.com/ScriptReference/MonoBehaviour.StartCoroutine.html)